tlf/id.go

```diff
@@ -144,21 +144,34 @@ func (id *ID) UnmarshalText(buf []byte) error {
 	return id.UnmarshalBinary(bytes)
 }
 
-// Type returns the type of TLF represented by this ID.
-//
-// Note that this function panics if the ID suffix is unknown, rather than
-// returning tlf.Unkonwn.
-func (id ID) Type() Type {
+// SafeType returns the type of TLF represented by this ID.  If the ID
+// isn't valid, it returns tlf.Unknown along with an error.
+func (id ID) SafeType() (Type, error) {
+	if len(id.id) == 0 {
```

Comment:
Make this `if len(id.id) != idByteLen {`? It would be rare, but a malicious would still be able to crash our servers by passing in a non-empty short ID.